src/components/Search/search-result.jsx

```diff
@@ -1,104 +1,94 @@
-import { Paragraph } from '@smallstep/step-ui';
+import { Heading } from '@smallstep/step-ui';
 import { makeStyles } from '@material-ui/styles';
 import {
   connectStateResults,
+  connectHits,
   Highlight,
-  Hits,
   Index,
   Snippet,
 } from 'react-instantsearch-dom';
-import { Box, ListItem, ListItemText } from '@material-ui/core';
+import { Box, Typography, ListItem, ListItemText } from '@material-ui/core';
 
-const useStyles = makeStyles({
+const useStyles = makeStyles((theme) => ({
   resultsBorder: {
-    borderColor: '#D3D3D3',
-    marginBottom: -16,
-    marginTop: -2,
+    borderBottom: `1px solid ${theme.palette.accent.grey3}`,
+  },
+  link: {
+    textDecoration: 'none',
+    color: 'inherit',
   },
   hits: {
-    marginLeft: -30,
-    marginRight: 10,
     '& mark ': {
-      backgroundColor: '#84A8FF',
-    },
-    '& ul ': {
-      listStyle: 'none',
-    },
-    '& a:-webkit-any-link ': {
-      textDecoration: 'none',
-      color: 'inherit',
+      backgroundColor: theme.palette.accent.yellow,
     },
   },
-});
+}));
 
 const HitCount = connectStateResults(({ searchResults }) => {
   const hitCount = searchResults && searchResults.nbHits;
-  const theme = useStyles();
+  const classes = useStyles();
   return hitCount > 0 ? (
-    <Box borderBottom={1} textAlign={'right'} className={theme.resultsBorder}>
-      <Box mr={2}>
+    <Box textAlign={'right'} className={classes.resultsBorder} mb={2}>
+      <Typography variant="body1">
         {hitCount} result{hitCount !== 1 ? `s` : ``}
-      </Box>
+      </Typography>
     </Box>
   ) : (
-    <Box
-      className={theme.resultsBorder}
-      minWidth={295}
-      textAlign={'left'}
-      ml={0.5}
-    >
-      There are no results found
+    <Box minWidth={295}>
+      <Typography variant="body2">There are no results found.</Typography>
     </Box>
   );
 });
 
-const PageHit = ({ hit }) => (
-  <Box mx={-2}>
+const PageHit = ({ hit }) => {
+  const classes = useStyles();
+
+  return (
     <Link
       to={
         hit.title[0] === hit.title[0].toUpperCase()
           ? `/docs/${hit.slug}`
           : `/docs/step-cli/reference/${hit.slug}`
       }
+      className={classes.link}
     >
-      <ListItem button attribute="slug" hit={hit} tagName="mark">
+      <ListItem button attribute="slug" hit={hit} className={classes.hits}>
         <ListItemText
           primary={
-            <h4>
+            <Heading variant="h6">
               <Highlight attribute="title" hit={hit} tagName="mark" />
-            </h4>
+            </Heading>
           }
           secondary={
-            <Paragraph>
+            <Typography variant="body2">
               <Snippet attribute="excerpt" hit={hit} tagName="mark" />
-            </Paragraph>
+            </Typography>
           }
         />
       </ListItem>
     </Link>
-  </Box>
+  );
+};
+
+const Hits = connectHits(({ hits }) =>
+  hits.map((hit) => <PageHit key={hit.slug} hit={hit} />)
```

Comment:
I found this thing from algolia where you can make your own custom `Hits` component instead of trying to target theirs with styles (weird negative margins and stuff).